### DISCUSSION TITLE
DBの接続情報にFormatDSN使うのはどうでしょうか?

### DISCUSSION THREAD

**Discussion by nnabeyang** [budougumi0617/go_todo_app#73] [Ideas]
次のようにFormatDSN使ったコードが紹介されるのかと思っていたのですが、fmt.Sprintfで結合されたコードだったので意外でした。あえてfmt.Sprintfで結合していると思うのですが、どういった理由でしょうか?
```go
func New(ctx context.Context, cfg *config.Config) (*sqlx.DB, func(), error) {
	c := mysql.Config{
		DBName:               cfg.DBName,
		User:                 cfg.DBUser,
		Passwd:               cfg.DBPassword,
		Addr:                 cfg.DBHost + ":" + cfg.DBPort,
		Net:                  "tcp",
		ParseTime:            true,
		AllowNativePasswords: true,
	}
       db, err := sql.Open("mysql", c.FormatDSN())
...
```

**Comment by budougumi0617** [comment]
ご指摘ありがとうございます（返信が遅くなり申し訳ありません。）
`mysql.Config` の存在を知らずに`fmt.Sprintf` を使って実装していたまま…というのが理由だったりします。
「こちらのほうがよいです」的な補足を正誤表に追加しておきます 🙏

**Reply by nnabeyang** [reply]
@budougumi0617 了解しました。ご回答いただきまして、ありがとうございます。

**Comment by budougumi0617** [comment]
対応がめちゃくちゃ遅くなってしまいました 🙇 正誤表に記載しました！
- https://github.com/budougumi0617/go_todo_app/pull/75
- https://github.com/budougumi0617/go_todo_app/releases/tag/v1.0.10
